compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -725,6 +725,9 @@ impl<'a, 'tcx> CastCheck<'tcx> {
                             },
                             // array-ptr-cast
                             Ptr(mt) => {
+                                if !mt.ty.is_sized(fcx.tcx, fcx.param_env) {
```

Comment:
This type may have region variables. That'll cause an ICE. Try using:

```suggestion
                                if !fcx.type_is_sized_modulo_regions(fcx.param_env, mt.ty) {
```

Reply:
Thanks! Didn't know this.